server/aesdsocket.c

```diff
@@ -0,0 +1,367 @@
+		close(socket_fd);
+		closelog();
+	    	exit(EXIT_FAILURE);
+	    }
+
+            if (chdir("/") == -1)   //change directory
+	    {
+            	syslog(LOG_ERR, "chdir error");
+            	closelog();
+            	close(socket_fd);
+            	return rc;
+            }
+
+    	     devnull_fd = open("/dev/null", O_RDWR);
+            dup2(devnull_fd, STDIN_FILENO);
+            dup2(devnull_fd, STDOUT_FILENO);
+            dup2(devnull_fd, STDERR_FILENO);
+            
+            //close file descriptors 
+            close(STDIN_FILENO);  
+            close(STDOUT_FILENO);
+            close(STDERR_FILENO);
+    	}
+    }
+   
+   
+   //listen to client connections 
+   if((status = listen(socket_fd,LISTEN_BACKLOG))<0)
+   {
+      syslog(LOG_ERR,"listen error\n");
+      return rc;
+   }
+    
+    read_buff = (char *) malloc(sizeof(char) * BUFFER_LEN);
+    write_buff = (char *) malloc(sizeof(char) *BUFFER_LEN );
+    
+    file_fd = open(FILE_PATH,O_RDWR | O_APPEND | O_CREAT,FILE_PERMISSIONS); //open file 
+   
+    if(file_fd==-1)
+     {
+      syslog(LOG_ERR,"Cannot open file\n");
+      return 0;
+     
+     }
+     
+     
+   
+    
+    while(1)
+    {
+         
+         
+      if((client_fd = accept(socket_fd,(struct sockaddr *)&client_addr,&client_addr_size))<0)  //accept connection from client socket
+      {
+        syslog(LOG_ERR,"accept connection failed\n");
+        return rc;
+      
+      }
+      
+     if ((client_ip = inet_ntoa(client_addr.sin_addr)) == NULL)    //obtain ip address of client
+     {
+        syslog(LOG_ERR,"client ip failed\n");
+        return rc;
+      
+     }
+     
+     else{
+     
+     syslog(LOG_DEBUG,"Accepted connection from %s\n", client_ip);
+     
+     }
+     
+     buf_loc = 0;
+     
+     //receive packets from client socket
+     
+	do 
+	{
+	    bytes_read = recv(client_fd, read_buff + buf_loc, BUFFER_LEN, 0);
+	    if(bytes_read == -1) 
+	    {  
+	       syslog(LOG_ERR,"recieve error\n");
+               free_memory();
+        	return rc;
+    	    }
+    	    
+	    ++realloc_val;
+	    
+	    buf_loc += bytes_read;
+	    
+            byte_total +=bytes_read;
+            
+            read_buff = (char *)realloc(read_buff, realloc_val*BUFFER_LEN*(sizeof(char))); //reallocate read buffer to make more space 
+            
+	    if(read_buff == NULL)
+	    {
+
+            	syslog(LOG_ERR,"realloc error\n");
+               free_memory();
+               return rc;
+            }
+
+	} while(strchr(read_buff, '\n') == NULL);
+	
+	
+	read_buff[buf_loc] = '\0';
+
+
+      
+        bytes_write = write(file_fd,read_buff, buf_loc); //write bytes to file 
+        
+        if(bytes_write == -1)
+        {
+            syslog(LOG_ERR,"write error\n");
+            free_memory();
+            return rc;
+        }
+
+	
+	write_buff = (char *) realloc(write_buff, byte_total*(sizeof(char))); //reallocate write buffer to make more space
```

Comment:
@chth2844 what assumption does this code make about the size of the available RAM heap relative to the size of all packets sent?